VLMs 2.0 blog (#2846)
huggingface/blog: vlms-2025.md

The model can take in long videos, PDFs, screenshots and more. It has agentic capabilities as well. 

### Smol yet Capable Models

The community used to scale intelligence through the number of parameters, and then high-quality synthetic data. After a certain point, the benchmarks saturated and scaling models had diminishing returns. The community went to shrink larger models through various methods, like distillation. This makes sense because it reduces compute costs, simplifies deployment, and unlocks use cases like local execution, enhancing data privacy. 

When we say small vision language models we often refer to models with less than 2B parameters that can be run on consumer GPUs. SmolVLM is a good example model family for smaller vision language models. Instead of shrinking larger models, went all the way and tried to fit models into tiny number of parameters like 256M, 500M and 2.2B. SmolVLM2, for instance, attempted to solve video understanding in these sizes and found 500M to be a good trade-off. At Hugging Face, we have built an iPhone application, HuggingSnap, to demonstrate that these model sizes can achieve video understanding on consumer devices. 

> Another striking model is [gemma3-1b-it](https://huggingface.co/google/gemma-3-1b-it) by Google DeepMind. It’s particularly exciting as it’s one of the smallest multimodal models to have 32k token context window, and support to 140+ languages. The model comes with Gemma 3 family of models, which the largest model was ranking first on LMArena. The largest model was then distilled to 1B variant. 

Reviewer: ```suggestion
Another striking model is [gemma3-1b-it](https://huggingface.co/google/gemma-3-1b-it) by Google DeepMind. It’s particularly exciting as it’s one of the smallest multimodal models to have 32k token context window, and supports 140+ languages. The model comes with the Gemma 3 family of models, with its largest model ranking first on Chatbot Arena at the time. The largest model was then distilled to a 1B variant. 
```

> Lastly, although not the smallest, [Qwen2.5-VL-3](https://huggingface.co/Qwen/Qwen2.5-VL-3B-Instruct)B-Instruct is worth noting. The model can do various tasks ranging from localization (object detection and pointing), document understanding, to agentic tasks, with context length up to 32k tokens.

Reviewer: ```suggestion
Lastly, although not the smallest, [Qwen2.5-VL-3B-Instruct](https://huggingface.co/Qwen/Qwen2.5-VL-3B-Instruct) is worth noting. The model can do various tasks ranging from localization (object detection and pointing), to document understanding, to agentic tasks; with context length up to 32k tokens.
```

> To get a nice understanding of MoEs we recommend reading [this fantastic article](https://huggingface.co/blog/moe).

### Vision-Language-Action Models

VLMs are even making their mark in the field of robotics\! There, they are known as Vision-language-action models (VLA). But don't be fooled, those are mainly VLMs with a little moustache and hat. VLAs take images and text instructions, and return text indicating actions for the robot to take directly. VLAs extend vision language models by adding action and state tokens to interact with and control physical environments. They are usually fine-tuned on top of a base VLM. Some people extend this definition further and define VLAs as any model interacting visually with a real or digital world. In this definition, VLAs can do UI navigation or be used in agentic workflows. But many people believe those applications fall in the VLM domain.

> These extra tokens represent the system’s internal state (how it perceives the environment), actions (what it does based on commands), and time-related information (like the order of steps in a task). These tokens are appended to the vision language input to generate actions or policy. 

Reviewer: ```suggestion
Extra tokens are used represent the system’s internal state (how it perceives the environment), actions (what it does based on commands), and time-related information (like the order of steps in a task). These tokens are appended to the vision language input to generate actions or policy. 
```

I think extra tokens were not mentioned before

### Object Detection, Segmentation, Counting with Vision Language Models

As we’ve seen in earlier sections, VLMs enable *generalization* over traditional computer vision tasks. Models can now take in images and a variety of prompts, such as open-ended text, and output structured text with localization tokens (for detection, segmentation and more).

Last year, [PaliGemma](https://huggingface.co/blog/paligemma) was the first model to attempt solving these tasks. The model takes in an image and text, where text is a description of an object of interest, along with a task prefix. The text prompt looks like “segment striped cat” or “detect bird on the roof”.

For detection, the model outputs the bounding box coordinates as *tokens*. For segmentation, on the other hand, the model outputs detection tokens and segmentation tokens. These segmentation tokens aren’t all the segmented pixel coordinates, but codebook indices that are decoded by a variational autoencoder trained to decode these tokens into valid segmentation masks (as shown in the figure below).

![PaliGemma3](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/vlm2/pg2-seg.png)

> Many models have been introduced to do localization tasks after PaliGemma. Late last year, an upgraded version of PaliGemma, PaliGemma2, appeared with the same capabilities and better performance. Another model that came after was Molmo by Allen AI, which can point to instances with dots and count object instances. 

Reviewer: ```suggestion
Many models have been introduced to do localization tasks after PaliGemma. Late last year, an upgraded version of PaliGemma, PaliGemma 2, appeared with the same capabilities and better performance. Another model that came later was Molmo by Allen AI, which can point to instances with dots and count object instances. 
```

![molmo](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/vlm2/molmo-pointing.png)

Qwen2.5-VL can also detect, point to, and count objects, and this includes UI elements as objects too\! 

![Qwen2.5VL](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/vlm2/qwen3-gui.png)

### Multimodal Safety Models

Vision language models in production require **filtering** inputs and outputs to prevent jailbreaks and harmful outputs for compliance. Harmful content varies from inputs with violence to sexually explicit content. That’s where multimodal safety models come in: they are used before and after vision language models to filter their inputs and outputs. They are just like LLM safety models but with additional image input.

> In early 2025, Google introduced the first open multimodal safety model, [ShieldGemma 2](https://huggingface.co/google/shieldgemma-2-4b-it). It is built on Shield Gemma, the text-only safety model. This model takes in images and content policies and returns whether an image is safe for a given policy. *Policy* refers to a criterion in which the image is inappropriate. ShieldGemma 2 can also be used to filter outputs of image generation models. 

Reviewer: ```suggestion
In early 2025, Google introduced the first open multimodal safety model, [ShieldGemma 2](https://huggingface.co/google/shieldgemma-2-4b-it). It is built on ShieldGemma, the text-only safety model. This model takes in images and content policies and returns whether an image is safe for a given policy. *Policy* refers to a criterion in which the image is inappropriate. ShieldGemma 2 can also be used to filter outputs of image generation models. 
```

Taken from [the paper](https://arxiv.org/pdf/2406.11251)

ColBERT-like models, like ColPali, are also dual encoder models, with a twist: ColPali has a vision language model as an image encoder, and a large language model as a text encoder. These models are inherently not encoders, but the models output embeddings, which are then passed to a “MaxSim”. The outputs are multiple vectors, one for each token, unlike DSE. In MaxSim, the similarity between each text token embedding and each image patch embedding is calculated, and this approach captures nuances better.  Due to this reason, ColBERT-like models are less cost-efficient, have better performance. 

Below you can see the indexing latency for ColPali. Since it’s just a single model, it’s also easier to maintain. 

![ColPali](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/vlm2/colpali.png) 
Taken from [the paper](https://arxiv.org/pdf/2407.01449)

On Hugging Face Hub, you can find these models under the task “[Visual Document Retrieval](https://huggingface.co/models?pipeline_tag=visual-document-retrieval&sort=trending)”. 

> The most popular benchmark for this task is ViDoRe benchmark, it consists of documents in **English** and **French**, with documents varying from financial reports, scientific figures to administrative documents. Each example of ViDoRe has the document as image, a query and potential answers. The documents matched with the queries help with contrastive pre-training, so the ViDoRe train set is used to train new models.

Reviewer: ```suggestion
The most popular benchmark for this task is ViDoRe, which consists of documents in **English** and **French**, with documents varying from financial reports, scientific figures to administrative documents. Each example of ViDoRe has the document as image, a query and potential answers. The documents matched with the queries help with contrastive pre-training, so the ViDoRe train set is used to train new models.
```

```
| Model Name               | Sizes                                         | Why we love it                                                 |
|--------------------------|-----------------------------------------------|----------------------------------------------------------------|
| Qwen2.5-VL               | from 3B to 72B                                | Great versatile model with agentic capabilities, math and more |
| RolmOCR                  | 7B                                            | Very performant OCR model                                      |
| Kimi-VL-Thinking                  | 16B MoE with 3B active parameters             | Best reasoning model                                           |
| SmolVLM2                 | 256M, 500M (our favorite!), 2.2B                              | Smallest video language model                                  |
| Llama 4 Scout & Maverick | 109B/400B MoE with 17B active parameters      | Loooooong context                                              |
| Molmo                    | 1B, 7B, 72B and MoE with 1B active parameters | Fully open model with localization capabilities on top                |

```

> Aaaaand that’s it! We hope you found this blog post useful to catch up with all the things that happened past year. We are looking forward to seeing all the things you’ll build with the models in this blog. 🤗 Below we provide some links on more in-depth explanations to each topic in this blog post. 

Reviewer: ```suggestion
Aaaaand that’s it! We hope you found this blog post useful to catch up with all the things that happened this past year. We are looking forward to seeing all the things you’ll build with the models in this blog. 🤗 Below we provide some links on more in-depth explanations to each topic in this blog post. 
```
